ion-schema/src/main/kotlin/com/amazon/ionschema/model/ConstraintId.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazon.ionschema.model
+
+/**
+ * Constraints are uniquely identified by their name (i.e. their ISL keyword). The [valueType] is included so that it is
+ * always possible to safely cast a constraint to the appropriate type, and so that constraint implementations cannot
+ * (easily) use the incorrect name or value type.
+ *
+ * Note—when creating an API, prefer to use [AnyConstraintId] instead of `TypedConstraintId<*>` so that clients of the
```

Comment:
`TypedConstraintId<*>` is a name that got changed as I was working on this. I guess I missed this comment. It should say something like this:
```suggestion
 * When creating APIs that accept `ConstraintId<*>` (specifically the wildcard typed `ConstraintId`), use this interface instead.
 * This interface exists because it is not allowed for an enum class to implement a generic interface, and so this allows [KnownConstraintIds] to implement `AnyConstraintId`.
```